test/integration/addcat_test.go

```diff
@@ -87,35 +91,38 @@ func RandomBytes(n int64) []byte {
 func DirectAddCat(data []byte, conf testutil.LatencyConfig) error {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	const numPeers = 2
 
 	// create network
-	mn, err := mocknet.FullMeshLinked(ctx, numPeers)
-	if err != nil {
-		return err
-	}
+	mn := mocknet.New(ctx)
 	mn.SetLinkDefaults(mocknet.LinkOptions{
 		Latency: conf.NetworkLatency,
 		// TODO add to conf. This is tricky because we want 0 values to be functional.
 		Bandwidth: math.MaxInt32,
 	})
 
-	peers := mn.Peers()
-	if len(peers) < numPeers {
-		return errors.New("test initialization error")
-	}
-
-	adder, err := core.NewIPFSNode(ctx, core.ConfigOption(MocknetTestRepo(peers[0], mn.Host(peers[0]), conf, core.DHTOption)))
+	adder, err := core.NewNode(ctx, &core.BuildCfg{
+		Online: true,
+		Host:   mock.MockHostOption(mn),
+	})
 	if err != nil {
 		return err
 	}
 	defer adder.Close()
-	catter, err := core.NewIPFSNode(ctx, core.ConfigOption(MocknetTestRepo(peers[1], mn.Host(peers[1]), conf, core.DHTOption)))
+
+	catter, err := core.NewNode(ctx, &core.BuildCfg{
+		Online: true,
+		Host:   mock.MockHostOption(mn),
+	})
 	if err != nil {
 		return err
 	}
 	defer catter.Close()
 
+	err = mn.LinkAll()
```

Comment:
One reason why this `err` assignment should be inside the if clause: it is not used elsewhere in the code except when it returns early here. It is analogous to the case `if _, _, err := aFunc(); err != nil`